scarb/src/core/manifest/mod.rs

```diff
@@ -141,3 +142,100 @@ pub fn edition_variant(edition: Edition) -> String {
     };
     edition
 }
+
+#[derive(Clone, Debug, Default)]
+pub struct FeaturesDefinition(BTreeMap<FeatureName, Vec<FeatureName>>);
+
+impl FeaturesDefinition {
+    pub fn new(features: BTreeMap<FeatureName, Vec<FeatureName>>) -> Self {
+        Self(features)
+    }
+
+    pub fn all(&self) -> Keys<'_, FeatureName, Vec<FeatureName>> {
+        self.0.keys()
+    }
+
+    pub fn get(&self, feature: &FeatureName) -> Option<&Vec<FeatureName>> {
+        self.0.get(feature)
+    }
+
+    pub fn contains_key(&self, feature: &FeatureName) -> bool {
+        self.0.contains_key(feature)
+    }
+
+    pub fn default_features(&self) -> Vec<FeatureName> {
+        self.0.get("default").cloned().unwrap_or_default()
+    }
+
+    pub fn select(&self, enabled: &[FeatureName], default_enabled: bool) -> SelectedFeatures {
```

Comment:
Nit:
```suggestion
    pub fn select(&self, enabled_features: &[FeatureName], default_enabled: bool) -> SelectedFeatures {
```
- is more readable arg name
- mirrors `not_found_features`